src/Qdrant.Client/QdrantClient.cs

```diff
@@ -225,7 +233,9 @@ public class QdrantClient : IDisposable
 		QuantizationConfig? quantizationConfig = null,
 		string? initFromCollection = null,
 		TimeSpan? timeout = null,
-		CancellationToken cancellationToken = default)
+		CancellationToken cancellationToken = default,
+		ShardingMethod? shardingMethod = null,
+		SparseVectorConfig? sparseVectorsConfig = null)
```

Comment:
_Suggestion:_ It's a general API convention to have [`CancellationToken` as the last parameter](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068) to async methods that support cancellation. Since this change will be binary compat breaking wherever the new parameters are placed, suggest moving `timeout` and `cancellationToken` to be the last parameters in all methods that have new parameters, for consistency.

The linked rule can be enforced through code analysis if it would be useful.

Reply:
My question is: do we break backward compatibility if I keep Cancellation Token as last arg? Because I added new fields and it may break users code because of another args order

Reply:
We are more concerned with API compatibility (e.g., I updated the client and I _don't_ have to change my code, it still works as-is).

But, with you mentioning that C# has named optional parameters (sorry, I barely know any C# 😂), I think it's fine to keep these as last parameters.

Reply:
True, we are worrying about API compatibility. But because of https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068, we have to place `CancellationToken` as the last argument, I will fix it. Thanks @russcam for pointing out this issue, nobody from the qdrant core team is dotnet backend developer, we didn't know =)

Reply:
No worries 🙂 If you wanted to not break binary compatibility at all when new methods are added to request photos, new methods can be added that accept the respective proto request type, and have method overloads call this. The exisiting methods remain as convenience methods and can be refactored to call the new method for each API, respectively. At the point where there's comfort in breaking binary compatibility (e.g. a semantic major version update), the convenience methods can be changed to expose new parameters.